_pages/fsa/dynamicforms/core/basic-form-validations.md

```diff
@@ -0,0 +1,285 @@
+  },
+  ....
+}
+```
+
+All arguments defined in JSON file represents inputs parameters for compareToCurrentDate function from DefaultFormValidators and they are added to the function in run-time. In our example based on mapping defined in configuration we will find following validation function:
+
+```typescript
+static compareToCurrentDate(operator) {
+   return (control: AbstractControl): ValidationErrors | null => {
+     const inputVal = new Date(control.value as string);
+     const today = new Date();
+     switch (operator) {
+       case 'shouldBeGreater':
+         return inputVal.getTime() > today.getTime()
+           ? null
+           : { InvalidDate: true };
+       case 'shouldBeLess':
+         return inputVal.getTime() < today.getTime()
+           ? null
+           : { InvalidDate: true };
+     }
+   };
+ }
+```
+and assign that function with input parameter 'shouldBeGreated' (operator in this case) to the corresponding field. In same way we can pass multiple input parameters via arguments attribute in case function requires more and add them to validation function respectively.
+
+## How to override existing validation in custom App
+
+In order to override some predefined validation, customers should be able to do that in their modules by referencing their custom functions with the same name as existing ones. Let's try to override compareToCurrentDate function, so that it just console logs customized message and returns always true for validation.  We will start in one of our modules by specifying following configuration:
+```typescript
+X-module.ts
+....
+      validators: {
+        compareToCurrentDate: {
+          validator: CategoryFormsModule.customFunction
+        }
+      }
+....
+```
+From code snippet above we can see that for validation with name compareToCurrentDate we have new function defined in CategoryFormsModule. Functions has following implementation:
+
+```typescript
+Custom validation
+static customFunction(regex) {
+  return (control: AbstractControl): ValidationErrors | null => {
+    console.log('Custom function works!');
+      return null;
+  };
+}
+```
+As described in introduction, function has one console log and always returns true for validation. In this particular case, JSON file should not be edited since name of new function is the same as for old one.
+
+## How to add new validation in custom App and invoke that from JSON
+
+Procedure for adding new function is more or less the same as for override. First we need to define new function somewhere and its implementation can be similar to the one for override. After that we only have to expose new function via configuration and both of those things are describe in code block below:
+```typescript
+New validation function
+static customNewFucntion(regex) {
+    return (control: AbstractControl): ValidationErrors | null => {
+      console.log('New function works!');
+        return null;
+    };
+  }
+ 
+....
```

Comment:
Same comment as before. It would be clearer to have the whole example with `ConfigModule.withConfig`...